usbsdmux/i2c_gpio.py

```diff
@@ -88,15 +70,72 @@ def set_pin_to_input(self, pins):
         Arguments:
         pins -- Combination of Pca9536.gpio_*
         """
-        self._directionMask = self._directionMask | pins
-        self._write_register(self._register_configuration, self._directionMask)
+        direction = self._read_register(self._register_configuration)
+        direction = direction | pins
+        self._write_register(self._register_configuration, direction)
+
+    def get_gpio_config(self):
+        """
+        Returns the state of the configuration register.
+        """
+        return self._read_register(self._register_configuration)
 
-    def output_values(self, values):
+    def get_input_values(self):
+        """
+        Reads the value currently present on the port from the input register.
+        """
+        return self._read_register(self._register_inputPort)
+
+    def output_values(self, values: int, bitmask: int = 0xFF):
         """
         Writes the given values to the GPIO-expander.
         Pins configured as Inputs are not affected by this.
 
         Arguments:
         values -- Combination of Pca9536.gpio_*
+        bitmask -- Only update bits in the register that are '1' in the bitmask
         """
-        self._write_register(self._register_outputPort, values)
+
+        if bitmask == 0xFF:
+            # trivial case: Let's just write the value
+            self._write_register(self._register_outputPort, values)
+        else:
+            # complex case: Let's do a read-modify-write
+            val = self._read_register(self._register_outputPort)
+            val = (val & ~bitmask) & 0xFF  # reset masked bits
+            val = val | (values & bitmask)  # set bits set in values and bitmask
+            self._write_register(self._register_outputPort, val)
+
+
+class Pca9536(I2cGpio):
+    """
+    Interface to control a Pca9536 that is connected to the auxiliary-I2C of a
+    Microchip USB2642.
+    """
+
+    # The PCA9536 I2C slave Address in 7-Bit Format
+    _I2cAddr = 0x41
+
+    gpio_0 = 0x01
+    gpio_1 = 0x02
+    gpio_2 = 0x04
+    gpio_3 = 0x08
+
+
+class Tca6408(I2cGpio):
+    """
+    Interface to control a TCA6408 that is connected to the auxiliary-I2C of a
+    Microchip USB2642.
+    """
+
+    # The TCA6408 I2C slave Address in 7-Bit Format
+    _I2cAddr = 0b0100_000
```

Comment:
We are getting into nit-picking territory here, but could you make both addresses `0b…` or `0x…`?
Also having the `_` separator 3 bits in makes sense in i²c 7bit-land but I'd prefer it at a nibble-boundary.